modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaServiceImplCheck.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+
+/**
+ * @author Kyle Miho
+ */
+public class UpgradeJavaServiceImplCheck
+	extends BaseAddComponentAnnotationCheck {
+
+	@Override
+	protected String getAnnotationContent(
+		String className, String content, JavaClass javaClass) {
+
+		return joinLines(
+			"@Component(", "\tproperty = {",
+			String.format(
+				"\t\t\"json.web.service.context.name=%s\",",
+				_getContextName(javaClass)),
```

Comment:
This one is actually tricky - we cannot reliably derive it from the class information. Rather, it is actually defined in the `service.xml` schema. This [logic in ServiceBuilder](https://github.com/liferay/liferay-portal/blob/master/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java?plain=1#L732-L748) defines the context variable that is then [lowercased into the template](https://github.com/liferay/liferay-portal/blob/master/modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_impl.ftl?plain=1#L26). The customer could have put _anything_ as the value for that.

The only way we will be fully accurate in this is to traverse to the `service.xml` file and mimic the ServiceBuilder logic.

Reply:
@drewbrokke It seems like we need to find the service.xml which is an arbitrary amount of directories above us, and I assume we find the first one. And then we prioritize using short-name attribute for the context name, and if that doesnt exist we need to use the namespace? Good find though!

Reply:
Yeah, the logic is:
1. Find `service.xml`
2. Try `portlet > short-name`
3. else try `namespace`

> after we filter the entityName since there can be multiple entities in 1 service.xml?

I think the elements we're inspecting are above any one entity.

Reply:
In portal, I see no usages of the `<portlet>` element, but a customer could conceivably still have that so we can check for it.

Reply:
Actually, an easier way would be to just rip off one of the existing other places where `portletShortName` is used. We can look around for them with: 
```sh
git grep 'portletShortName' -- ':/**/portal-tools-service-builder/*.ftl'
```

Reply:
Seems like looking for `public class ${portletShortName}PersistenceConstants {` would be the easiest.